compiler/optimizations/remoteValueForwarding.cpp

```diff
@@ -205,10 +205,25 @@ static bool canForwardValue(Map<Symbol*, Vec<SymExpr*>*>& defMap,
                             ArgSymbol*                    arg) {
   bool retval = false;
 
+  // Forward array values and references to array values.
+  // This is OK because the array/domain/distribution wrapper
+  // records have fields that do not vary.
+  // It does not matter if the on-body synchonizes.
+  // (The array class, e.g. DefaultRectangularArr, is what varies,
+  //  and it contains a pointer to the actual data, which might
+  //  be replaced with another pointer).
+  // An alternative strategy would be to migrate the contents of the
+  // array header class into the wrapper record - but that would require
+  // quite a lot of code changes, and some other features have entangled
+  // designs (including privitazation and the DSI interface).
+  if (isRecordWrappedType(arg->getValType())) {
+    retval = true;
```

Comment:
I'm pretty confident in it. isRecordWrappedType returns true for array, domain, distribution (and nothing else). The getValType() is necessary in this context since the argument we're interested in forwarding is a reference.

Array, domain, and distribution all work in RVF for the same reasons.